Tests/CodegenCLITests/VersionCheckerTests.swift

```diff
@@ -186,6 +186,41 @@ class VersionCheckerTests: XCTestCase {
     expect(result).to(equal(.versionMismatch(cliVersion: Constants.CLIVersion, apolloVersion: apolloVersion)))
   }
 
+  func test__matchCLIVersionToApolloVersion__givenPackageResolvedFileInXcodeWorkspace_withVersion2FileFormat_hasMatchingVersion_returns_versionMatch() throws {
+    // given
+    try fileManager.createFile(
+      body: version2PackageResolvedFileBody(apolloVersion: Constants.CLIVersion),
+      named: "Package.resolved",
+      inDirectory: "MyProject.xcworkspace/xcshareddata/swiftpm"
+    )
+
+    // when
+    let result = try VersionChecker.matchCLIVersionToApolloVersion(
+      projectRootURL: fileManager.directoryURL
+    )
+
+    // then
+    expect(result).to(equal(.versionMatch))
+  }
+
+  func test__matchCLIVersionToApolloVersion__givenPackageResolvedFileInXcodeWorkspace_withVersion2FileFormat_hasNonMatchingVersion_returns_versionMismatch() throws {
+    // given
+    let apolloVersion = "1.0.0.test-1"
+    try fileManager.createFile(
+      body: version2PackageResolvedFileBody(apolloVersion: apolloVersion),
+      named: "Package.resolved",
+      inDirectory: "MyProject.xcworkspace/xcshareddata/swiftpm"
+    )
+
+    // when
+    let result = try VersionChecker.matchCLIVersionToApolloVersion(
+      projectRootURL: fileManager.directoryURL
+    )
+
+    // then
+    expect(result).to(equal(.versionMismatch(cliVersion: Constants.CLIVersion, apolloVersion: apolloVersion)))
+  }
+
```

Comment:
There should also be a test that creates both an `xcworkspace` and `xcproject` file to ensure that the workspace is prioritized over the project file.

Reply:
Resolved in https://github.com/apollographql/apollo-ios/pull/3048/commits/9a7a1a09e1d708e1dbe0af8a0be5d695ffbaac1c.